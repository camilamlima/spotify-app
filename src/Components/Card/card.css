.btn-green{
    margin-bottom: 40px;
    background-color: #1DB954;
    border-radius: 50px;
    color: #fff;
    font-weight: 700;
}
.warp-img{
    height: 240px;
    overflow: hidden;
}
.card-img-top{
    width: 100%;
}