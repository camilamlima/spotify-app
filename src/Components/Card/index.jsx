import React, {Component} from 'react'

import './card.css'

class Card extends Component {

    render(){
      return(
        <div className="card">
          <div className="warp-img">
            <img className="card-img-top" src={this.props.artista.image} alt={this.props.artista.name} />
          </div>
          <div className="card-body">
            <h5 className="card-title">{this.props.artista.name}</h5>
            <a href={this.props.artista.href} className="btn btn-green" target="blank">Playlist</a>
          </div>
        </div>
      );
    }
}
export default Card;







