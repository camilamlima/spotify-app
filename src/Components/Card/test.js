import React from 'react';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Card from '.';

test('Testar componente de Card da aplicação', () => {

    let artista = {
        name: "Alcymar Monteiro",
        href: "https://open.spotify.com/artist/0TcVnvKse98awlZxtUKIOk",
        image: "https://i.scdn.co/image/1d0ccd0c4b5ec14c2c4575fd11f365883c971b0b"
    }

    const component = renderer.create(
      <BrowserRouter>
        <Card artista={artista}/>
      </BrowserRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});