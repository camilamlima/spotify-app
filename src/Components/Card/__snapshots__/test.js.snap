// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testar componente de Card da aplicação 1`] = `
<div
  className="card"
>
  <div
    className="warp-img"
  >
    <img
      alt="Alcymar Monteiro"
      className="card-img-top"
      src="https://i.scdn.co/image/1d0ccd0c4b5ec14c2c4575fd11f365883c971b0b"
    />
  </div>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Alcymar Monteiro
    </h5>
    <a
      className="btn btn-green"
      href="https://open.spotify.com/artist/0TcVnvKse98awlZxtUKIOk"
      target="blank"
    >
      Playlist
    </a>
  </div>
</div>
`;
